### PR TITLE
[v2.0.0] Add mocha unit test runner 

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
Adds [`mocha`](https://mochajs.org/) and [`chai`](https://www.chaijs.com/) testing framework dependencies to run unit tests within the project. We also add a `test` npm script which executes the tests with mocha via babel.

### Note
To ensure we have parity with the source files within the project, we are writing are unit tests in TypeScript and transpiling them with babel before passing to mocha.

Annoyingly we had to create a wrapper around [`@babel/register`](https://babeljs.io/docs/en/babel-register) to pass the `.ts.` extension for this to behave correctly. See this issue for further detail: https://github.com/babel/babel/issues/8673

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "main": "dist/lib.js",
   "license": "MIT",
   "scripts": {
-    "test": "",
+    "test": "mocha -r ./babel-register.js 'src/**/*.spec.ts'",
```
Comment (psbanka) [inline package.json]: Jest automatically handles splitting tests across processes, collecting and aggregating coverage. Also has nice built-in watch options. I think this might be a better choice for the project.

File: src/worker.spec.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { describe, it } from "mocha";
+import { expect } from "chai";
+
+import * as worker from "./worker";
```
[the remark this answers is not in the document]
Comment (clight-fastly) [reply src/worker.spec.ts]: We've "outlawed" default exports in our repo, I think it's a good decision.
https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html

File: package.json (hunk 1 above)
Comment (phamann) [reply package.json]: Agreed.

**Comment by phamann** [comment]
Am going to close this PR in favour of using Jest as our test framework.
